client/components/dot-pager/index.jsx

```diff
@@ -90,22 +90,29 @@ export const DotPager = ( {
 		if ( ! hasDynamicHeight ) {
 			return;
 		}
+		if ( 1 === numPages ) {
+			return;
+		}
 
 		const targetHeight = pagesRef.current?.querySelector( '.is-current' )?.offsetHeight;
+
 		setPagesStyle( targetHeight ? { height: targetHeight } : undefined );
-	}, [ hasDynamicHeight, setPagesStyle ] );
+	}, [ hasDynamicHeight, setPagesStyle, numPages ] );
 
 	useEffect( () => {
 		updateLayout();
 	}, [ currentPage, updateLayout ] );
 
 	useEffect( () => {
+		if ( 1 === numPages ) {
+			return;
+		}
 		window.addEventListener( 'resize', updateLayout );
 
 		return () => {
 			window.removeEventListener( 'resize', updateLayout );
 		};
-	}, [ updateLayout ] );
+	}, [ updateLayout, numPages ] );
```

Comment:
The goal here is to `updateLayout` only when the `hasDynamicHeight && numPages > 1` condition is true. The whole thing could be separated to its own hook:
```js
function useUpdateLayout( enabled, currentPage, updateLayout ) {
	// save callback to a ref so that it doesn't need to be a dependency of other hooks
	const savedUpdateLayout = useRef();
	useEffect( () => {
		savedUpdateLayout.current = updateLayout;
	}, [ updateLayout ] );

	// fire when the `currentPage` parameter changes
	useEffect( () => {
		if ( ! enabled ) {
			return;
		}

		savedUpdateLayout.current();
	}, [ enabled, currentPage ] );

	// fire when the window resizes
	useEffect( () => {
		if ( ! enabled ) {
			return;
		}

		const onResize = () => savedUpdateLayout.current();
		window.addEventListener( 'resize', onResize );
		return () => window.removeEventListener( 'resize', onResize );
	}, [ enabled ] );
}
```
and used as:
```js
const updateEnabled = hasDynamicHeight && numPages > 1;

useUpdateLayout( updateEnabled, currentPage, () => {
	const targetHeight = pagesRef.current?.querySelector( '.is-current' )?.offsetHeight;
	setPagesStyle( targetHeight ? { height: targetHeight } : undefined );
} );
```
Because we can't call hooks conditionally, we need to pass the `enabled` parameter and do nothing in the effects when it's `false`.

Reply:
I like Jarda's idea of abstracting that logic to a separate hook a lot. Makes it all much more readable, it was kinda bothering me before as well.

Reply:
Thanks for the review @jsnajdr  and @tyxla . I implemented the suggested hook. The code looks way cleaner now.